### ISSUE TITLE
Can't train animals

### ISSUE THREAD

**Issue by SpartanDRJJ** [sumghai/Replimat#35]
I found someone else had this problem but that was in 2019 and supposedly fixed and my log is different. Anyways here's the log:

Patty threw exception in WorkGiver Train: System.NullReferenceException: Object reference not set to an instance of an object
  at CommonSense.IngredientPriority+FoodUtility_FoodOptimality.Postfix (System.Single& __result, Verse.Pawn eater, Verse.Thing foodSource, Verse.ThingDef foodDef, System.Single dist, System.Boolean takingToInventory) [0x00028] in <3ecde58463224f83b01ce35df5423d54>:0 
  at (wrapper dynamic-method) RimWorld.FoodUtility.DMD<DMD<FoodOptimality_Patch2>?22041472::FoodOptimality_Patch2>(Verse.Pawn,Verse.Thing,Verse.ThingDef,single,bool)
  at (wrapper dynamic-method) RimWorld.FoodUtility.DMD<DMD<SpawnedFoodSearchInnerScan_Patch1>?-636133632::SpawnedFoodSearchInnerScan_Patch1>(Verse.Pawn,Verse.IntVec3,System.Collections.Generic.List`1<Verse.Thing>,Verse.AI.PathEndMode,Verse.TraverseParms,single,System.Predicate`1<Verse.Thing>)
  at (wrapper dynamic-method) RimWorld.FoodUtility.DMD<DMD<BestFoodSourceOnMap_Patch4>?-596410496::BestFoodSourceOnMap_Patch4>(Verse.Pawn,Verse.Pawn,bool,Verse.ThingDef&,RimWorld.FoodPreferability,bool,bool,bool,bool,bool,bool,bool,bool,bool,bool,RimWorld.FoodPreferability)
  at RimWorld.WorkGiver_InteractAnimal.TakeFoodForAnimalInteractJob (Verse.Pawn pawn, Verse.Pawn tamee) [0x0001c] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at (wrapper dynamic-method) RimWorld.WorkGiver_Train.DMD<DMD<JobOnThing_Patch2>?643441664::JobOnThing_Patch2>(RimWorld.WorkGiver_Train,Verse.Pawn,Verse.Thing,bool)
  at RimWorld.WorkGiver_Scanner.HasJobOnThing (Verse.Pawn pawn, Verse.Thing t, System.Boolean forced) [0x00000] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at RimWorld.JobGiver_Work+<>c__DisplayClass3_1.<TryIssueJobPackage>b__0 (Verse.Thing t) [0x00013] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at Verse.GenClosest+<>c__DisplayClass2_0.<ClosestThingReachable>b__0 (Verse.Thing t) [0x00034] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at Verse.GenClosest.<ClosestThing_Global>g__Process|5_0 (Verse.Thing t, Verse.GenClosest+<>c__DisplayClass5_0& ) [0x00047] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at Verse.GenClosest.ClosestThing_Global (Verse.IntVec3 center, System.Collections.IEnumerable searchSet, System.Single maxDistance, System.Predicate`1[T] validator, System.Func`2[T,TResult] priorityGetter) [0x0008d] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at Verse.GenClosest.ClosestThingReachable (Verse.IntVec3 root, Verse.Map map, Verse.ThingRequest thingReq, Verse.AI.PathEndMode peMode, Verse.TraverseParms traverseParams, System.Single maxDistance, System.Predicate`1[T] validator, System.Collections.Generic.IEnumerable`1[T] customGlobalSearchSet, System.Int32 searchRegionsMin, System.Int32 searchRegionsMax, System.Boolean forceAllowGlobalSearch, Verse.RegionType traversableRegionTypes, System.Boolean ignoreEntirelyForbiddenRegions) [0x0014e] in <4489f1367c1c4d76b2ae1272c8ed691d>:0 
  at ExpandedRoofing.ClosestThingReachableHelper.ClosestThingReachableWrapper (Verse.IntVec3 root, Verse.Map map, Verse.ThingRequest thingReq, Verse.AI.PathEndMode peMode, Verse.TraverseParms traverseParams, System.Single maxDistance, System.Predicate`1[T] validator, System.Collections.Generic.IEnumerable`1[T] customGlobalSearchSet, System.Int32 searchRegionsMin, System.Int32 searchRegionsMax, System.Boolean forceGlobalSearch, Verse.RegionType traversableRegionTypes, System.Boolean ignoreEntirelyForbiddenRegions) [0x00049] in <f814eb4228dc4aa0a801e6bf24509816>:0 
  at (wrapper dynamic-method) RimWorld.JobGiver_Work.DMD<DMD<TryIssueJobPackage_Patch0>?-231181184::TryIssueJobPackage_Patch0>(RimWorld.JobGiver_Work,Verse.Pawn,Verse.AI.JobIssueParams)
Verse.Log:Error(String, Boolean)
RimWorld.JobGiver_Work:DMD<DMD<TryIssueJobPackage_Patch0>?-231181184::TryIssueJobPackage_Patch0>(JobGiver_Work, Pawn, JobIssueParams)
Verse.AI.ThinkNode_PrioritySorter:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Tagger:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Subtree:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Conditional:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn, JobIssueParams)
Verse.AI.Pawn_JobTracker:DMD<DMD<DetermineNextJob_Patch4>?1728232960::DetermineNextJob_Patch4>(Pawn_JobTracker, ThinkTreeDef&)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:DMD<DMD<EndCurrentJob_Patch1>?-988020736::EndCurrentJob_Patch1>(Pawn_JobTracker, JobCondition, Boolean, Boolean)
Verse.AI.Pawn_JobTracker:JobTrackerTick()
Verse.Pawn:DMD<DMD<Tick_Patch2>?775612800::Tick_Patch2>(Pawn)
Verse.TickList:Tick()
Verse.TickManager:DMD<DMD<DoSingleTick_Patch1>?-796596736::DoSingleTick_Patch1>(TickManager)
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()

**Comment by sumghai** [comment]
Please provide additional troubleshooting information:

- What Operating System and version are you running?
- What RimWorld game version (including 32/64-bit) are you running?
- What Mods are you running?
- Have you narrowed down the specific mods you suspect to be conflicting with Replimat?
- Does your problem also occur if you only run a test savegame with only Core and Replimat?

**Comment by SpartanDRJJ** [comment]
Sorry for the late reply. My internet went down 5 minutes after your reply and it was 3AM so I went to bed.

Anyways I did find the problem. It was the Animal Food Restrictions mod and in hindsight I'm not sure this mod was related to the issue. I thought it was since my google search brought up a similar error but looking through the log again I realize there's not even a mention of replimat. 

Hope you didn't spend much time on this.

So in case you were wondering what the actual problem was I don't quite know why it happened only what mod caused it. The animals WERE allowed kibble and that is what the trainers use. But when I enabled all food for the animals suddenly the trainers would pick up kibble and train the animals even though they were already allowed it before aswell. ooh well ¯\_(ツ)_/¯

Have a good one.
